Fixed open issues #1 and #2

bluez-sh opened the pull request:
Added a url builder method that takes an input from the user for username and builds the url. Also, made the code PEP8 compliant.

bluez-sh committed:
Fixed open issues #1 and #2

sibasish14 reviewed (changes requested):
@Swaraj1998 What if the entered username doesn't exist?
You should add a condition for that.

bluez-sh commented:
Okay...right on it.

bluez-sh committed:
Update scrape.py

bluez-sh commented:
@sibasish14 thanx for pointing it out :+1:

palash25 commented:
@Swaraj1998 @sibasish14  so happy to see that you guys are actively contributing. I'm quite busy these days and might not have the time to review and merge these but don't stop coding. Meanwhile explore other projects by other mentors too and also other organizations and also keep updating the spreadsheet.

sibasish14 reviewed (approved):
ack https://github.com/palash25/CoWaPS/pull/8/commits/d7ff9b7abedc9d9b0fc7ce66cf437d6f514733b1

palash25 commented on scrape.py:
```diff
@@ -2,26 +2,45 @@
 
 
+def url_builder():
+    uname = raw_input("Enter user name: ")
```
Lets change `uname` to `username`
You also need to change all the `raw_input()` to `input()` as it is no longer supported by python3

bluez-sh replied:
kk...will do it.

palash25 commented on scrape.py:
```diff
@@ -2,26 +2,45 @@
+        print "Username not found!"
+
+        while 1:
+            ch = raw_input("Do you want to try again? (y/n): ")
```
Lets change this `while` block. If the user provides an invalid username then the program should exit. A better approach would be to put `user_info` soup in a `try-except` block and display the message *Username does not exist, please try again*

bluez-sh replied:
Yeah, that works too....i'll change it.

palash25 reviewed (commented):
@sibasish14 don't ack commits without asking me :wink:

sibasish14 commented:
@palash25 will keep that in mind

bluez-sh committed:
Update scrape.py

pep8speaks commented:
Hello @Swaraj1998! Thanks for updating the PR.



Cheers ! There are no PEP8 issues in this Pull Request. :beers: 

##### Comment last updated on December 19, 2017 at 13:15 Hours UTC

bluez-sh committed:
Update scrape.py

bluez-sh commented:
@palash25 I tried to use try-except but it didn't work. It never went to the except block. Looks like it is not an exception and `user_info` just turns out to be empty. So, I just changed it to `if not user_info:`.

bluez-sh commented:
@palash25 Also, should I take the user name as a command line argument or should I let it be taken as input inside the script?

palash25 commented:
@Swaraj1998 command line argument

palash25 commented:
can you upload a screenshot of the output in your PR description

sibasish14 commented on scrape.py:
```diff
@@ -2,26 +2,36 @@
+    
+    if not user_info:
+        print("Username does not exist, please try again. ")
+        return
```
@Swaraj1998 if you are returning `None`, then I don't think you should say `please try again` because it simply exits from the program and there's no "trying again". You should probably change it to `Username doesn't exist, exiting...... `.

@Swaraj1998 @palash25 would like to know your opinions.

bluez-sh replied:
@sibasish14 Well, "try again" could mean  "run the script again". But, anyway, either way works. I'll change it.

palash25 commented on scrape.py:
```diff
@@ -2,26 +2,36 @@
+
+
 def fetch(url):
     r = requests.get(url)
```
I think it would be better to check if the user account exist over here instead of traversing through the code below. Just check what kind of status code the requests object is returning using if-else

bluez-sh replied:
@palash25 Yeah, that's a good idea. I'll check for the status code.

bluez-sh replied:
@palash25 Okay, I made the changes. Have a look at the code.

bluez-sh committed:
Update scrape.py

Username is now taken as a command line argument. Also, existence of username is checked by status code of the response.

palash25 commented:
@Swaraj1998  add a screenshot of your output

bluez-sh commented:
@palash25 Here you go. ![screenshot from 2017-12-19 22-43-10](https://user-images.githubusercontent.com/20240824/34169710-4191bbe4-e50e-11e7-9cab-1c87e96b95b1.png)

palash25 commented:
Nicely done :+1: @Swaraj1998

bluez-sh commented:
@palash25 Thanx :)
